runtime/rasdump/javadump.cpp

```diff
@@ -4603,10 +4648,21 @@ JavaCoreDumpWriter::writeThread(J9VMThread* vmThread, J9PlatformThread *nativeTh
 #else /* defined(J9ZOS390) || defined(J9ZTPF) */
 		_OutputStream.writeCharacters("3XMTHREADINFO3           No native callstack available for this thread\n");
 #endif /* defined(J9ZOS390) || defined(J9ZTPF) */
-		_OutputStream.writeCharacters("NULL\n");
 	}
+#if JAVA_SPEC_VERSION >= 21
+	if ((NULL != vmThread) && (NULL != vmThread->threadObject)) {
+		_OutputStream.writeCharacters("3XMTHREADINFO4           Type:");
+		if (IS_JAVA_LANG_VIRTUALTHREAD(vmThread, vmThread->threadObject)) {
+			_OutputStream.writeCharacters("Virtual, java/lang/CarrierThread:");
+			_OutputStream.writePointer(vmThread->carrierThreadObject);
+			_OutputStream.writeCharacters("\n");
+		} else {
+			_OutputStream.writeCharacters("Platform\n");
+		}
+	}
+#endif /* JAVA_SPEC_VERSION >= 21 */
 
-	_OutputStream.writeCharacters("NULL\n");
+	_OutputStream.writeCharacters("NULL\nNULL\n");
```

Comment:
Why?

Reply:
Previously there are 1 or 2 `NULL` line depending on if a Native call stack is printed or not. I added the extra `NULL` line to match the case where 2 line will be printed

Reply:
The only place where I can find two consecutive `NULL` lines is at the end of the `SHARED CLASSES` section, and I don't think that is appropriate.

Reply:
ok, I will remove the extra NULL line.